packages/smithy-http/src/smithy_http/endpoints.py

```diff
@@ -13,11 +16,23 @@ class Endpoint(interfaces.Endpoint):
     headers: interfaces.Fields = field(default_factory=Fields)
 
 
+class EndpointResolutionError(SmithyException):
+    """Exception type for all exceptions raised by endpoint resolution."""
```

Comment:
nit: I'd prefer that all exceptions go in a central `exceptions.py`

Reply:
Makes sense.  Moved.